internal/v2/remote_signer/remote_signer.go

```diff
@@ -14,13 +14,14 @@ type remoteSigner struct {
 	leafCert *x509.Certificate
 	cstream s2av2pb.S2AService_SetUpSessionClient
 	localIdentity *commonpbv1.Identity
+	signatureAlgorithm s2av2pb.SignatureAlgorithm
```

Comment:
I don't think we can know the signatureAlgorithm in advance - we need to figure it out on every call to the `Sign` API. Since we're still not sure how to proceed with that (I think?), I would suggest the following for this PR: change the `getSignatureAlgorithm` method to accept the signerOpts and have it return a static value for now. WDYT?

Reply:
Sounds good! I think there may be a way to get the signature algorithm info from the ClientHello information, if there is, we can make the change in the next PR.